### ISSUE TITLE
AttributeError: 'Series' object has no attribute 'nonzero'

### ISSUE THREAD

**Issue by mtvector** [CHPGenetics/GMM-demux#2]
Hi there,
I'm having an issue with running GMM-demux, I installed with pip, I've also tried installing pandas 1.1 or pandas 0.25 and have the same issue.

```
==============================GMM-Demux Initialization==============================
Confidence threshold: 0.8
Output directory: SSD_mtx
Traceback (most recent call last):
  File "/wynton/home/ye/mschmitz1/utils/miniconda3/envs/multiseq/bin/GMM-demux", line 8, in <module>
    sys.exit(main())
  File "/wynton/home/ye/mschmitz1/utils/miniconda3/envs/multiseq/lib/python3.8/site-packages/GMM_Demux/GMM_Demux.py", line 116, in main
    purified_df = classify_drops.purify_droplets(GMM_full_df, confidence_threshold)
  File "/wynton/home/ye/mschmitz1/utils/miniconda3/envs/multiseq/lib/python3.8/site-packages/GMM_Demux/classify_drops.py", line 144, in purify_droplets
    drop_idx = data_df.index[((data_df["Confidence"] < confidence_threshold) | (data_df["Cluster_id"] == 0)).nonzero()[0]]
  File "/wynton/home/ye/mschmitz1/utils/miniconda3/envs/multiseq/lib/python3.8/site-packages/pandas/core/generic.py", line 5130, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'Series' object has no attribute 'nonzero'


Thanks!
M

```

**Comment by xhongyi** [comment]
Hi,

Could you please provide a full list of your package versions? You can obtain it with `pip3 freeze`

Also may I ask which OS are you using? If you are using macOS or linux, make sure you install GMM-Demux with pip3 instead of pip. In some systems the default pip is pip3 but in other cases it's python2-pip. Could you check with pip --version

It should be something like this: `pip 20.0.2 from /usr/lib/python3/dist-packages/pip (python 3.8)`. Notice python3.8 in the parentheses.

**Comment by xhongyi** [comment]
It might be an update of pandas that removed the nonzero attribute (removed since the pandas 0.25 update). I'll test and fix it tonight!

Thanks,
Hongyi

**Comment by xhongyi** [comment]
I have fixed the problem and tested it. I've submitted the new version (0.2.1.2) to pip.

Try install GMM-Demux again (first remove it with `pip3 uninstall GMM-demux`) and it should be fixed.

Let me know if it is resolved and I'll close the ticket.

Thanks for using GMM-Demux!

**Comment by mtvector** [comment]
Yeah that works!

**Comment by mtvector** [comment]
@xhongyi Also in your `store_simplified_classify_result` function there's a fix needed!

Traceback (most recent call last):
  File "/wynton/home/ye/mschmitz1/utils/miniconda3/envs/multiseq/bin/GMM-demux", line 8, in <module>
    sys.exit(main())
  File "/wynton/home/ye/mschmitz1/utils/miniconda3/envs/multiseq/lib/python3.8/site-packages/GMM_Demux/GMM_Demux.py", line 113, in main
    classify_drops.store_simplified_classify_result(GMM_full_df, class_name_ary, args.simplified, sample_num, confidence_threshold)
  File "/wynton/home/ye/mschmitz1/utils/miniconda3/envs/multiseq/lib/python3.8/site-packages/GMM_Demux/classify_drops.py", line 115, in store_simplified_classify_result
    MSM_idx = data_df.index[(data_df["Cluster_id"] > sample_num).nonzero()[0]]
  File "/wynton/home/ye/mschmitz1/utils/miniconda3/envs/multiseq/lib/python3.8/site-packages/pandas/core/generic.py", line 5130, in __getattr__
    return object.__getattribute__(self, name)
AttributeError: 'Series' object has no attribute 'nonzero'

**Comment by xhongyi** [comment]
Yep. Fixed them with v0.2.1.3. Sorry about the inconvenience! Hope that it works for you now!

Hongyi
